shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -38,7 +38,9 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
     val fileContents = os.read(file)
     val allLines = fileContents.splitSane('\n').toList
     val strw = new java.io.StringWriter
-    val out = new java.io.PrintWriter(strw)
+    val out = new java.io.PrintWriter(strw) {
+      override def println(): Unit = print('\n')
```

Comment:
Though I'd add a comment to this effect:
```suggestion
      override def println(): Unit = print('\n') // to avoid inserting CRLF on Windows
```